plugins/pdr_deterministic_plugin/ufm_sim_web_service/isolation_mgr.py

```diff
@@ -271,18 +276,23 @@ def check_link_down_condition(self, port_obj, port_name, ports_counters):
     
     def read_next_set_of_high_ber_or_pdr_ports(self, endpoint_port):
         issues = {}
-        ports_counters = self.ufm_client.get_telemetry(endpoint_port, Constants.PDR_DYNAMIC_NAME)
+        ports_counters = self.ufm_client.get_telemetry(endpoint_port, Constants.PDR_DYNAMIC_NAME,self.test_mode)
         if ports_counters is None:
             # implement health check for the telemetry instance
             self.logger.error("Couldn't retrieve telemetry data")
             return issues
         for index, row in ports_counters.iterrows():
             port_name = f"{row.get('port_guid', '').split('x')[-1]}_{row.get('port_num', '')}"
+            if self.test_mode:
+                # I want to add them once we get our first telemetry.
+                if not self.ports_data.get(port_name):
+                    self.ports_data[port_name] = PortData(port_name)
             port_obj = self.ports_data.get(port_name)
```

Comment:
move this line do be after 
port_name =

if you will not get port object - no need to do all the flow for test mode

Reply:
I do not follow, it is after port_name =
and it before port_obj = , because we need to insert data there because there are none, because it is test_mode

Reply:
I see, so OK.